contracts/strategies/SnowballStrategyV1.sol

```diff
@@ -0,0 +1,235 @@
+        snowGlobe.approve(address(stakingContract), MAX_UINT);
+    }
+
+    function deposit(uint amount) external override {
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
+        require(DEPOSITS_ENABLED == true, "SnowballStrategyV1::_deposit");
+        if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+            uint unclaimedRewards = checkReward();
+            if (unclaimedRewards > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+                _reinvest(unclaimedRewards);
+            }
+        }
+        require(depositToken.transferFrom(msg.sender, address(this), amount));
+        uint lpAmountDeposited = _stakeDepositTokens(amount);
+        _mint(account, getSharesForDepositTokens(lpAmountDeposited));
+        totalDeposits = totalDeposits.add(lpAmountDeposited);
+        emit Deposit(account, lpAmountDeposited);
+    }
+
+    function _stakeDepositTokens(uint amount) private returns (uint) {
+        require(amount > 0, "SnowballStrategyV1::_stakeDepositTokens");
+        snowGlobe.deposit(amount);
+        uint sLPAmount = snowGlobe.balanceOf(address(this));
+        uint lpAmountDeposited = sLPAmount.mul(snowGlobe.getRatio()).div(1e18);
+        stakingContract.deposit(sLPAmount);
+        return lpAmountDeposited;
+    }
+
+    function withdraw(uint amount) external override {
+        uint depositTokenAmount = getDepositTokensForShares(amount);
+        if (depositTokenAmount > 0) {
+            _withdrawDepositTokens(depositTokenAmount);
+            _safeTransfer(address(depositToken), msg.sender, amount);
+            _burn(msg.sender, amount);
+            totalDeposits = totalDeposits.sub(depositTokenAmount);
+            emit Withdraw(msg.sender, depositTokenAmount);
+        }
+    }
+
+    function _withdrawDepositTokens(uint amount) private {
+        require(amount > 0, "SnowballStrategyV1::_withdrawDepositTokens");
+        uint sharesAmount = amount.mul(1e18).div(snowGlobe.getRatio());
+        stakingContract.withdraw(sharesAmount);
+        snowGlobe.withdraw(sharesAmount);
+    }
+
+    function reinvest() external override onlyEOA {
+        uint unclaimedRewards = checkReward();
+        require(unclaimedRewards >= MIN_TOKENS_TO_REINVEST, "SnowballStrategyV1::reinvest");
+        _reinvest(_convertRewardIntoWAVAX(unclaimedRewards));
+    }
+
+    function _convertRewardIntoWAVAX(uint pendingReward) private returns (uint) {
+        stakingContract.getReward();
+        DexLibrary.swap(
+            rewardToken.balanceOf(address(this)),
```

Comment:
gas efficiency to use `pendingReward` instead of `rewardToken.balanceOf(address(this))`